server/events/vcs/github_client.go

```diff
@@ -430,16 +481,106 @@ func (g *GithubClient) UpdateStatus(ctx context.Context, request types.UpdateSta
 }
 
 // [WENGINES-4643] TODO: Move the checks implementation to UpdateStatus once github checks is stable
-// UpdateChecksStatus updates the status check
 func (g *GithubClient) UpdateChecksStatus(ctx context.Context, request types.UpdateStatusRequest) error {
-	// TODO: Implement updating github checks
-	// - Get all checkruns for this SHA
-	// - Match the UpdateReqIdentifier with the check run. If it exists, update the checkrun. If it does not, create a new check run.
+	result, _, err := g.client.Checks.ListCheckRunsForRef(ctx, request.Repo.Owner, request.Repo.Name, request.Ref, &github.ListCheckRunsOptions{})
+	if err != nil {
+		return err
+	}
 
-	// Checks uses Status and Conlusion. Need to map models.CommitStatus to Status and Conclusion
-	// Status -> queued, in_progress, completed
-	// Conclusion -> failure, neutral, cancelled, timed_out, or action_required. (Optional. Required if you provide a status of "completed".)
-	return nil
+	for _, checkRun := range result.CheckRuns {
+		if *checkRun.Name == request.StatusName {
+			return g.UpdateCheckRun(ctx, request, *checkRun.ID)
+		}
+	}
+
+	return g.CreateCheckRun(ctx, request)
```

Comment:
`CreateCheckRun` and `UpdateCheckRun` should be private methods. Unless you're planning on using these outside?

Reply:
Done!